### PR TITLE
Improve thisdd4hep.sh setup scripts

### PR DESCRIPTION
**Pull request by hansenjo**
BEGINRELEASENOTES
- Improve path handling in setup scripts, by avoiding duplicates in paths. Also support whitespaces in paths and don't add system paths to library paths on macOS

ENDRELEASENOTES

### CODE REVIEW

**Comment by hansenjo** [comment]
I'm checking why the CI checks are failing with this. The script works fine on my private machines.

**Comment by hansenjo** [comment]
This seems to be working now in the CI environment as well. The PR is ready to be reviewed.

File: CMakeLists.txt [hunk 1]
```diff
@@ -58,6 +58,12 @@ IF(${CMAKE_CXX_STANDARD} LESS 14)
   MESSAGE(FATAL_ERROR "DD4hep requires at least CXX Standard 14 to compile")
 ENDIF()
 
+# ----for APPLE scripts have to set the DYLD_LIBRARY_PATH
+if(APPLE)
+  set(USE_DYLD 1)
+  set(CMAKE_MACOSX_RPATH 1)
```
Comment (petricm) [inline CMakeLists.txt:64]: `set(CMAKE_MACOSX_RPATH 1)` is set in `cmake/DD4hepBuild.cmake` along with all the rpath treatment. And `USE_DYLD` could be drooped entirely in favor of using the variable `APPLE`

**Review by petricm — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (hansenjo) [reply CMakeLists.txt]: Sure, that works. Requested changes made in commit d926c9692b5d9aacb96585ba756b7b42794c6115

**Pull request merged**